presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ApplyNode.java

```diff
@@ -92,7 +92,15 @@ public ApplyNode(
      */
     public boolean isResolvedScalarSubquery()
     {
-        return isScalar(subquery) && subqueryAssignments.values().stream()
+        return isScalar(subquery) && isResolvedSubquery();
+    }
+
+    /**
+     * @return true when subquery output symbols are directly mapped to ApplyNode output symbols
+     */
+    public boolean isResolvedSubquery()
+    {
+        return subqueryAssignments.values().stream()
```

Comment:
I don't think non-scalar subquery `ApplyNode`, which directly map subquery symbols to outputs are algebraically correct. How could you use such non-scalar subquery symbol in projection for instance?